libs/application-generic/src/usecases/tier-restrictions-validate/tier-restrictions-validate.usecase.ts

```diff
@@ -79,37 +79,33 @@ export class TierRestrictionsValidateUsecase {
     return [];
   }
 
-  private async getMaxDelayInMs(command: TierRestrictionsValidateCommand, organization: OrganizationEntity) {
-    const isPackagesQ1Enabled = await this.is4PackageTierActivated(command, organization);
-    const featureFlags = { [FeatureFlagsKeysEnum.IS_2025_Q1_TIERING_ENABLED]: isPackagesQ1Enabled };
-
-    return getFeatureForTierAsNumber(
-      FeatureNameEnum.PLATFORM_MAX_DIGEST_WINDOW_TIME,
-      organization.apiServiceLevel || ApiServiceLevelEnum.FREE,
-      featureFlags,
-      true
-    );
-  }
-
-  private async isOrganizationExcludedFromRestriction(
+  private async getMaxDelayInMs(
     command: TierRestrictionsValidateCommand,
-    organization: OrganizationEntity
+    organization: OrganizationEntity,
+    stepType: StepTypeEnum.DELAY | StepTypeEnum.DIGEST
   ) {
-    return await this.featureFlagsService.getFlag({
-      key: FeatureFlagsKeysEnum.IS_TIER_DURATION_RESTRICTION_EXCLUDED_ENABLED,
-      defaultValue: false,
-      environment: { _id: command.environmentId },
-      organization,
-    });
-  }
+    const config = getFlagNameByType(stepType);
 
-  private async is4PackageTierActivated(command: TierRestrictionsValidateCommand, organization: OrganizationEntity) {
-    return await this.featureFlagsService.getFlag({
-      key: FeatureFlagsKeysEnum.IS_2025_Q1_TIERING_ENABLED,
-      defaultValue: false,
+    const systemLimit = await this.featureFlagsService.getFlag({
+      key: config.system,
+      defaultValue: SYSTEM_LIMITS.DEFER_DURATION_MS,
       environment: { _id: command.environmentId },
       organization,
     });
+
+    // If the system limit is not the default, we need to use it as the absolute limit for special cases instead of the tier limit
+    const isSpecialLimit = systemLimit !== SYSTEM_LIMITS.DEFER_DURATION_MS;
```

Comment:
Ditto.

Reply:
same here i wanted to state that it is a special case